protobuf_serialization/reader.nim

```diff
@@ -42,7 +43,53 @@ proc readFieldInto[T: object](
       raise (ref ValueError)(msg: "not enough bytes")
     memoryInput(tmp).readValueInternal(value)
 
-proc readFieldInto[T: not object and (seq[byte] or not seq)](
+proc readFieldInto[K, V](
```

Comment:
let's hide this behind a conformance test suite compile flag for now - it's not something that should make it into the "public" api before it's ready.

the long term plan here would be to ensure that we can support "external" type mappings like this easily, so that users can write their own "extensions", while at the same time, the [well-known](https://protobuf.dev/reference/protobuf/google.protobuf/) types can be supported via `import pbs/wellknown/datetime` (or something like that) etc